packages/tpetra/CMakeLists.txt

```diff
@@ -260,21 +281,28 @@ ELSE ()
   ENDIF () # Whether we are cross compiling
 ENDIF () # Whether we have CUDA and MPI
 
-# Sanity checks for Tpetra_ASSUME_CUDA_AWARE_MPI_DEFAULT.
+# Sanity checks for Tpetra_ASSUME_GPU_AWARE_MPI_DEFAULT.
 
-IF (NOT (DEFINED Tpetra_ASSUME_CUDA_AWARE_MPI_DEFAULT))
-  MESSAGE (FATAL_ERROR "Failed to set Tpetra_ASSUME_CUDA_AWARE_MPI_DEFAULT!")
+IF (NOT (DEFINED Tpetra_ASSUME_GPU_AWARE_MPI_DEFAULT))
+  MESSAGE (FATAL_ERROR "Failed to set Tpetra_ASSUME_GPU_AWARE_MPI_DEFAULT!")
 ENDIF ()
 
-IF (NOT Tpetra_ENABLE_CUDA AND Tpetra_ASSUME_CUDA_AWARE_MPI_DEFAULT)
-  MESSAGE (FATAL_ERROR "Tpetra_ASSUME_CUDA_AWARE_MPI_DEFAULT is ON, but CUDA is not enabled!")
+IF (NOT Tpetra_ENABLE_GPU AND Tpetra_ASSUME_GPU_AWARE_MPI_DEFAULT)
+  MESSAGE (FATAL_ERROR "Tpetra_ASSUME_GPU_AWARE_MPI_DEFAULT is ON, but a GPU build is not enabled!")
 ENDIF ()
 
+TRIBITS_ADD_OPTION_AND_DEFINE(
+  Tpetra_ASSUME_GPU_AWARE_MPI
+  TPETRA_ASSUME_GPU_AWARE_MPI
+  "Assume that the MPI implementation that Tpetra uses is GPU aware (via HIP or CUDA).  \"GPU aware\" means that MPI can use a GPU device allocations (the result of cuda/hipMalloc) as send and receive buffers.  If MPI is GPU aware, then Tpetra can give GPU device allocations directly to MPI sends and receives.  MPI implementations might not support all MPI communication operations.  For example, different versions of OpenMPI support different subsets of MPI operations.  Tpetra developers may safely assume that MPI_*[Ss]end and MPI_*[Rr]ecv work.  Anything more than that should be considered specific to the MPI implementation."
+  ${Tpetra_ASSUME_GPU_AWARE_MPI_DEFAULT})
+
+# This is for backward compatability
 TRIBITS_ADD_OPTION_AND_DEFINE(
   Tpetra_ASSUME_CUDA_AWARE_MPI
   TPETRA_ASSUME_CUDA_AWARE_MPI
   "Assume that the MPI implementation that Tpetra uses is CUDA aware.  \"CUDA aware\" means that MPI can use CUDA device allocations (the result of cudaMalloc) as send and receive buffers.  If MPI is CUDA aware, then Tpetra can give CUDA device allocations directly to MPI sends and receives.  MPI implementations might not support all MPI communication operations.  For example, different versions of OpenMPI support different subsets of MPI operations (see https://www.open-mpi.org/faq/?category=runcuda ).  Tpetra developers may safely assume that MPI_*[Ss]end and MPI_*[Rr]ecv work.  Anything more than that should be considered specific to the MPI implementation."
```

Comment:
"CUDA aware" -> "GPU-aware"

Reply:
Isn't this for backward compatibility, as stated in the comment?